### ISSUE TITLE
can't pickle module objects

### ISSUE THREAD

**Issue by yangziyi1990** [keras-team/keras#14546]
**I write a model but can not save it:

The model as follows:**

        input1 = Input(shape=(None, Input_size1, ), name='input1')
        lstm_dialogue = Bidirectional(GRU(self.hidden_size,
                                          return_sequences=True,
                                          dropout=0.2,
                                          recurrent_dropout=0.2), name='BiGRU')(input1)

        dense_dialogue = TimeDistributed(Dense(self.hidden_size, activation='tanh'), name='dense1')(lstm_dialogue)
        attention = Dense(1, activation='tanh')(dense_dialogue)     # try diff act
        attention = TimeDistributed(Flatten(), name='flatten')(attention)
        attention = TimeDistributed(Activation('softmax'), name='activation')(attention)  # try different activations
        attention = TimeDistributed(RepeatVector(self.hidden_size), name='repeatvector')(attention)
        attention = TimeDistributed(Permute([2, 1]), name='permute')(attention)
        dialogue_representation = Multiply()([dense_dialogue, attention])
        dialogue_attention = Lambda(lambda xin: K.sum(xin, axis=1), output_shape=(None, self.hidden_size,))(dialogue_representation)
        x_out = TimeDistributed(Dense(1, activation='sigmoid'), name='dense2')(dialogue_attention)
        model = Model(inputs=input1, outputs=x_out)

**when i save the model, the error occur:**
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/main.py", line 113, in main
    run(argv)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/main.py", line 102, in run
    args.func.run(**args_dict)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/command/train.py", line 76, in run
    verbose=verbose
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/models/wrapper.py", line 71, in fit
    return self.model.fit(train_X_list, y, validation_X_list=validation_X_list, validation_y_list=validation_y, **merged_params)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/models/rnn.py", line 707, in fit
    print(train_model.to_json())
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/keras/engine/network.py", line 1212, in to_json
    model_config = self._updated_config()
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/keras/engine/network.py", line 1176, in _updated_config
    config = self.get_config()
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/site-packages/keras/engine/network.py", line 931, in get_config
    return copy.deepcopy(config)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 240, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/yangziyi/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 215, in _deepcopy_list
    append(deepcopy(a, memo))
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 240, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 240, in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 220, in _deepcopy_tuple
    y = [deepcopy(a, memo) for a in x]
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 220, in <listcomp>
    y = [deepcopy(a, memo) for a in x]
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 150, in deepcopy
    y = copier(x, memo)
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 220, in _deepcopy_tuple
    y = [deepcopy(a, memo) for a in x]
  File "/Users/yangziyi/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 220, in <listcomp>
    y = [deepcopy(a, memo) for a in x]
  File "/Users/opt/anaconda3/envs/python3.6/lib/python3.6/copy.py", line 169, in deepcopy
    rv = reductor(4)

**Comment by sushreebarsa** [comment]
@yangziyi1990 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
